module/VuFind/src/VuFind/Db/Table/Feedback.php

```diff
@@ -0,0 +1,156 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Db_Table
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\Db\Table;
+
+use Laminas\Db\Adapter\Adapter;
+use Laminas\Paginator\Paginator;
+use VuFind\Db\Row\RowGateway;
+
+/**
+ * Class Feedback
+ *
+ * @category VuFind
+ * @package  Db_Table
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class Feedback extends Gateway
+{
+    /**
+     * Constructor
+     *
+     * @param Adapter         $adapter Database adapter
+     * @param PluginManager   $tm      Table manager
+     * @param array           $cfg     Laminas configuration
+     * @param RowGateway|null $rowObj  Row prototype object (null for default)
+     * @param string          $table   Name of database table to interface with
+     */
+    public function __construct(
+        Adapter $adapter,
+        PluginManager $tm,
+        $cfg,
+        ?RowGateway $rowObj = null,
+        $table = 'feedback'
+    ) {
+        parent::__construct($adapter, $tm, $cfg, $rowObj, $table);
+    }
+
+    /**
+     * Get feedback by filter
+     *
+     * @param string|null $formName Form name
+     * @param string|null $siteUrl  Site URL
+     * @param string|null $status   Current status
+     * @param string|null $page     Current page
+     * @param int         $limit    Limit per page
+     *
+     * @return Paginator
+     */
+    public function getFeedbackByFilter(
+        ?string $formName = null,
+        ?string $siteUrl = null,
+        ?string $status = null,
+        ?string $page = null,
+        int $limit = 20
+    ): Paginator {
+        $sql = $this->getSql();
+        $select = $sql->select()->columns(
+            [
+                '*',
+                'user_name' => new \Laminas\Db\Sql\Expression(
+                    "CONCAT_WS(' ', u.firstname, u.lastname)"
+                ),
+                'manager_name' => new \Laminas\Db\Sql\Expression(
+                    "CONCAT_WS(' ', m.firstname, m.lastname)"
+                ),
+            ]
+        );
+        if (null !== $formName) {
+            $select->where->equalTo('form_name', $formName);
+        }
+        if (null !== $siteUrl) {
+            $select->where->equalTo('site_url', $siteUrl);
+        }
+        if (null !== $status) {
+            $select->where->equalTo('status', $status);
+        }
+        $select->join(['u' => 'user'], 'u.id = feedback.user_id', [], 'left')
```

Comment:
Might be better to use the constant $select::JOIN_LEFT instead of the hard-coded string 'left' ... though it's not a huge matter of importance since hopefully this will all get replaced by Doctrine in the fairly near future.

Reply:
You are right, using constant is better, I've changed it.

Reply:
You are right, using constant is better, I've changed it.